src/components/pages/PeoplePage/PeopleTable/PeopleTable.tsx

```diff
@@ -0,0 +1,201 @@
+  const centuries = searchParams.getAll('centuries') || [];
+  const sortBy = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+
+  const findParent = useCallback((personName: string | null) => {
+    const parent = people.find(person => person.name === personName);
+
+    if (parent) {
+      return (
+        <PersonLink person={parent} />
+      );
+    }
+
+    return personName || '-';
+  }, [people]);
+
+  useEffect(() => {
+    const filteredAndSortedPeople = people.filter(person => {
+      if (sex) {
+        return person.sex === sex;
+      }
+
+      return person;
+    }).filter(person => {
+      if (query) {
+        const lowerQuery = query.toLowerCase();
+        const personNames = person.name
+              + person.fatherName + person.motherName;
+
+        return personNames.toLowerCase().includes(lowerQuery);
+      }
+
+      return person;
+    }).filter(person => {
+      if (centuries.length) {
+        const personCentury = Math.ceil(person.born / 100);
+
+        return centuries.includes(personCentury.toString());
+      }
+
+      return person;
+    }).sort((prev, current) => {
+      switch (sortBy) {
+        case 'sex':
+        case 'name':
+          return prev[sortBy].localeCompare(current[sortBy]);
+        case 'born':
+        case 'died':
+          return prev[sortBy] - current[sortBy];
+        default:
+          return 0;
+      }
+    });
+
+    if (order) {
+      filteredAndSortedPeople.reverse();
+    }
+
+    setFilteredPeople(filteredAndSortedPeople);
+  }, [searchParams]);
+
+  return (
+    <table
+      data-cy="peopleTable"
+      className={'table is-striped is-hoverable'
+        + 'is-narrow is-fullwidth'}
+    >
+      {filteredPeople.length
+        ? (
+          <>
+            <thead>
+              <tr>
+                <th>
+                  <span className="is-flex is-flex-wrap-nowrap">
+                    Name
+                    <SearchLink
+                      params={{
+                        sort: !sortBy || !order ? 'name' : null,
+                        order: sortBy && !order ? 'desc' : null,
+                      }}
+                    >
+                      <span className="icon">
+                        {sortBy !== 'name' && <i className="fas fa-sort" />}
+                        {sortBy === 'name'
+                          && !order && <i className="fas fa-sort-down" />}
+                        {sortBy === 'name'
+                          && order && <i className="fas fa-sort-up" />}
+                      </span>
+                    </SearchLink>
+                  </span>
+                </th>
+
+                <th>
+                  <span className="is-flex is-flex-wrap-nowrap">
+                    Sex
+                    <SearchLink
+                      params={{
+                        sort: !sortBy || !order ? 'sex' : null,
+                        order: sortBy && !order ? 'desc' : null,
```

Comment:
Create constants for `!sortBy || !order` and `sortBy && !order` and reuse em everywhere